### PR TITLE
Support interfaces and abstract methods in them.

### PR DESCRIPTION
**Pull request by mprobst**

### CODE REVIEW

File: test/DartTest.ts [hunk 1]
```diff
@@ -113,6 +113,9 @@ describe('transpile to dart', () => {
       expectTranslate('class X extends Y implements Z {}')
           .to.equal(' class X extends Y implements Z { }');
     });
+    it('supports abstract methods', () => {
+      expectTranslate('interface X { x(); }').to.equal(' abstract class X { abstract x ( ) ; }');
```
Comment (alexeagle) [inline test/DartTest.ts]: opt: we should add to the e2e test to exercise the subtyping/subclassing relations and how they work at runtime

**Comment by alexeagle** [comment]
curious: did you find anything in the angular codebase which required this?

File: main.ts [hunk 2]
```diff
@@ -446,6 +450,14 @@ class Translator {
         this.visitFunctionLike(funcExpr);
         break;
 
+      case ts.SyntaxKind.MethodSignature:
+        var methodSignatureDecl = <ts.FunctionLikeDeclaration>node;
+        this.emit('abstract');
+        if (methodSignatureDecl.modifiers) this.visitEach(methodSignatureDecl.modifiers);
```
Comment (alexeagle) [inline main.ts]: maybe visitEach should short-circuit if the argument is falsy

Comment (mprobst) [inline main.ts]: Yeah. I'll add a maybeVisitEach for the cases where stuff is nullable. I don't like methods that are too acceptant of their args, so having a strict and a maybe variant might be nice.

**Pull request merged**